### PR TITLE
Fixes compilation failures on MacOSX 10.10

### PR DESCRIPTION
**Pull request by SamJBarney**
* Fixes https://github.com/cuberite/cuberite/issues/2327
* Replace old c-style casts with c++ casts
* Added `-Wno-error=old-style-cast` to Protocol18x.cpp

### CODE REVIEW

File: src/Generating/BioGen.cpp [hunk 1]
```diff
@@ -641,8 +641,8 @@ void cBioGenMultiStepMap::BuildTemperatureHumidityMaps(int a_ChunkX, int a_Chunk
 	// Re-map into integral values in [0 .. 255] range:
 	for (size_t idx = 0; idx < ARRAYCOUNT(a_TemperatureMap); idx++)
 	{
-		a_TemperatureMap[idx] = std::max(0, std::min(255, (int)(128 + TemperatureMap[idx] * 128)));
-		a_HumidityMap[idx]    = std::max(0, std::min(255, (int)(128 + HumidityMap[idx]    * 128)));
+		a_TemperatureMap[idx] = std::max(0, std::min(255, static_cast<int>(128 + TemperatureMap[idx] * 128)));
+		a_HumidityMap[idx]    = std::max(0, std::min(255, static_cast<int>(128 + HumidityMap[idx]    * 128)));
```
Comment (madmaxoft) [inline src/Generating/BioGen.cpp]: This could use the `Clamp` function instead of `max(min())`

Comment (SamJBarney) [inline src/Generating/BioGen.cpp]: I'll open an issue to deal with it later.

**Comment by SamJBarney** [comment]
Can someone pull this down and test world generation for me? It doesn't have to be a mac. I'm getting very blocky terrain with it.

**Comment by SamJBarney** [comment]
Found the issue. Two misplaced parens.

**Comment by SamJBarney** [comment]
Tested against another version. Generation is exactly the same. This is ready to merge if anyone wants to pass off on it.

File: src/Generating/BioGen.cpp [hunk 2]
```diff
@@ -429,7 +429,7 @@ void cBioGenMultiStepMap::InitializeBiomeGen(cIniFile & a_IniFile)
 	m_MushroomIslandSize  =        a_IniFile.GetValueSetI("Generator", "MultiStepMapMushroomIslandSize", m_MushroomIslandSize);
 	m_RiverCellSize       =        a_IniFile.GetValueSetI("Generator", "MultiStepMapRiverCellSize",      m_RiverCellSize);
 	m_RiverWidthThreshold =        a_IniFile.GetValueSetF("Generator", "MultiStepMapRiverWidth",         m_RiverWidthThreshold);
-	m_LandBiomesSize      = (float)a_IniFile.GetValueSetI("Generator", "MultiStepMapLandBiomeSize",      (int)m_LandBiomesSize);
+	m_LandBiomesSize      = static_cast<float>(a_IniFile.GetValueSetI("Generator", "MultiStepMapLandBiomeSize",      static_cast<int>(m_LandBiomesSize)));
```
Comment (tigerw) [inline src/Generating/BioGen.cpp]: Doesn't really matter, but maybe remove all alignment spaces with this line since it won't align anyway?

**Comment by hallucino** [comment]
I think your changes are only for the command line ? Because I have others errors with Xcode.

**Comment by SamJBarney** [comment]
Yes. Only for commandline. I was going to leave the Xcode ones to you since you had already started on that.

**Comment by SamJBarney** [comment]
So is no one merging this just because I haven't fixed the spacing that tigerw mentioned?

**Comment by tigerw** [comment]
Okay okay, so accusatory :P

I think it's mainly because everyone seems busy. My comment can be fixed later, so looks good.

**Comment by SamJBarney** [comment]
Sorry, I wasn't trying to come off accusatory. I just wanted to get on to fixing my git so I can actually rebase when I work.

**Comment by SamJBarney** [comment]
Well, work on more than one thing at a time.

**Comment by tigerw** [comment]
I said it jokingly anyhow.

> fixing my git
Indeed.

**Pull request merged**